File: 20171014/Aeneascryptographicdisc.cpp
#include <bits/stdc++.h>

using namespace std;
const long double PI = 3.141592653589793238L;
long double degree[266];
int d;
long double cal(long double x, long double y){
    long double dis = min(fabs(x - y), 360 - fabs(x - y));
    return d*sqrt(2 - 2*cos(dis*PI / 180));
}
int main() {
    ios_base::sync_with_stdio(false);
    cin >> d;
    string s;
    for(int i = 0; i< 26; i++){
        cin >> s;
        cin >> degree[(int)s[0]];
    }
    char last = 0;
    long double ans = d;
    while(cin >> s){
        for(auto&c:s) if(isalpha(c)){
            if(isalpha(last)){
                ans += cal(degree[toupper(last)], degree[toupper(c)]);
            }
            last = c;
        }
    }
    cout << ceil(ans);
}

File: 20171014/Aeneascryptographicdisc.py
import math
d = float(input())
degree = dict()
for i in range(26):
    c, d = input().split()
    degree[c] = float(d)
ans = d
s = ''.join(c for c in input() if c.isalpha()).upper()
def cal(x,y):
    dis = min(math.fabs(x - y), 360 - math.fabs(x - y))
    return d*math.sqrt(2. - 2.*math.cos(dis*math.pi/180))

for i in range(1,len(s)):
    ans += cal(degree[s[i-1]],degree[s[i]])
print(math.ceil(ans))